torch/csrc/api/src/optim/lbfgs.cpp

```diff
@@ -10,167 +10,602 @@
 #include <cmath>
 #include <functional>
 #include <vector>
+#include <algorithm>
 
 namespace torch {
 namespace optim {
 
-LBFGSOptions::LBFGSOptions(double learning_rate)
-    : learning_rate_(learning_rate) {}
+LBFGSOptions::LBFGSOptions(double lr) : lr_(lr) {}
 
-Tensor LBFGS::gather_flat_grad() {
+bool operator==(const LBFGSOptions& lhs, const LBFGSOptions& rhs) {
+  return (lhs.lr() == rhs.lr()) &&
+         (lhs.max_iter() == rhs.max_iter()) &&
+         (lhs.max_eval() == rhs.max_eval()) &&
+         (lhs.tolerance_grad() == rhs.tolerance_grad()) &&
+         (lhs.tolerance_change() == rhs.tolerance_change() &&
+         (lhs.history_size() == rhs.history_size()));
```

Comment:
Should we also check `line_search_fn` is the same?

Reply:
oops missed that. thanks for the catch!